File: javascript/9-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
 var isPalindrome = function(x) {
    if (x < 0) return false;
    let t = x, r = 0, m = 0;
    while (t > 0) {
        m = t % 10;
        r = r * 10 + m;
        t = Math.floor(t/10);
    }
    return r === x;
};

File: golang/9-palindrome-number.go
package main

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	t, r, m := x, 0, 0
	for x > 0 {
		m = x % 10
		r = r*10 + m
		x = x / 10
	}
	return r == t
}
